ensenso_camera/src/virtual_object_handler.cpp

```diff
@@ -25,9 +27,119 @@ namespace {
         buffer << file.rdbuf();
         return buffer.str();
     }
+
+    /// Class responsible for publishing NxLib virtual objects as ROS visualization_msgs::MarkerArray
+    struct VirtualObjectMarkerPublisher
+    {
+      VirtualObjectMarkerPublisher(const std::string &topic, double publishRate, const NxLibItem& objects,
+                                   const std::string &frame, std::atomic_bool& stop_)
+        : rate(publishRate), stop(stop_)
+      {
+        // Create output topic
+        ros::NodeHandle node;
```

Comment:
I am not quite sure if it is okay to destroy the local nodehandle after exciting the constructor of the `VirtualObjectMarkerPublisher`, after assigning a publisher to it etc.
As far as I know, this is okay, if atleast one nodehandle lives outside of this class (e.g. the NodeHandle of the camera itself).
I guess it would be also okay to make this NodeHandle a member of the `VirtualObjectMarkerPublisher` struct.
Correct me If I am wrong.

Reply:
Agreed, I'll update.